nibabel/tests/test_testing.py

```diff
@@ -171,12 +171,14 @@ def test_assert_re_in(regex, entries):
 
 
 def test_test_data():
-    assert test_data() == data_path
-    assert test_data() == os.path.abspath(
+    assert str(test_data()) == str(data_path)
+    assert str(test_data()) == os.path.abspath(
         os.path.join(os.path.dirname(__file__), '..', 'tests', 'data')
     )
     for subdir in ('nicom', 'gifti', 'externals'):
-        assert test_data(subdir) == os.path.join(data_path[:-10], subdir, 'tests', 'data')
+        assert str(test_data(subdir)) == os.path.join(
+            data_path.parent.parent, subdir, 'tests', 'data'
```

Comment:
Is `data_path` a `Path` instance? If so, wouldn't it be more natural to use the `/` operator instead of `os.path.join`? Also, maybe the assertions could use the `samefile` method?

Reply:
I added fa5f9207fbc1bee9e39bac865c80afb6987e13e1 to address this. LMK what you think.

Reply:
LGTM. 
I might, at some point in the future, if you don't think it's a bad idea, make a refactoring PR with a bunch of more explicit variable naming suggestions, e.g., here my instinct would be to do something like:
```python
expected = data_path.parent.parent / subdir / 'tests' / 'data'
assert str(test_data(subdir)) == expected
```

Reply:
You're more than welcome to.